gateway_test.go

```diff
@@ -1029,3 +1030,48 @@ func TestProxyUserAgent(t *testing.T) {
 		}
 	}
 }
+
+func buildAndLoadAPI(apiGens ...func(spec *APISpec)) {
```

Comment:
I'd make this use a temporary AppPath directory instead - we already do this in other tests. It's the best way to ensure that nothing else is in the dir, and that even if we panic, we won't leave anything behind. And you can delete with a single `os.RemoveAll` call.